calyx/src/frontend/parser.rs

```diff
@@ -582,26 +590,53 @@ impl CalyxParser {
         ))
     }
 
+    fn invoke_external_arg(input: Node) -> ParseResult<(ir::Id, ir::Id)> {
+        Ok(match_nodes!(
+            input.into_children();
+            [identifier(outcell), identifier(incell)] => (outcell, incell)
+        ))
+    }
+
+    fn invoke_external_args(input: Node) -> ParseResult<Vec<(ir::Id, ir::Id)>> {
+        Ok(match_nodes!(
+            input.into_children();
+            [invoke_external_arg(args)..] => args.collect(),
+        ))
+    }
+
+    fn ivk_extern_args_in_paren(
```

Comment:
Once you make `invoke_external_arg` and `invoke_external_args` into silent rules, you can remove the top two functions and inline their handling into `ivk_extern_args_in_paren`.

Reply:
After carefully looking at parser.rs and syntax.pest, I decided that the more efficient way may be to delete ivk_extern_args_in_paren and just make invoke_external_args represent the full invoke syntax including the square brackets. ivk_extern_args_in_paren is a rule that I made up initially to wrap the vectors produced by invoke_external_args in options. But now since we are not using options for external cells anymore, there is no need for that extra rule.